lib/helper/Puppeteer.js

```diff
@@ -851,6 +853,50 @@ class Puppeteer extends Helper {
     return proceedClick.call(this, locator, context, { waitForNavigation: true });
   }
 
+  /**
+   * {{> ../webapi/downloadFile }}
+   */
+  async downloadFile(locator) {
+    let fileName;
+    await this.page.setRequestInterception(true);
+    this.click(locator);
+
+    const xRequest = await new Promise((resolve) => {
+      this.page.on('request', (request) => {
+        fileName = request.url().split('/')[request.url().split('/').length - 1];
```

Comment:
I have some concerns about the filename.
Let's look into 2 scenarios:

1. File name is generated and there is no way to obtain it. How can we access that file in next steps of a test? 
2. File name is constant. Each next test overwrites the previous test. What if at one run file was not downloaded but test passes because, the file was left from previous run? 

I propose to generate a file name based on a template passed as second parameter.

```js
I.downloadFile('.download', 'avatar_{id}.jpg');
```

A method should return a complete file name so it could be used in a test with FileSystem helper:

```js
let filename = await I.downloadFile('.export', 'users_{id}.txt');
I.seeFile(filename);
I.seeInThisFile('Davert');
I.seeInThisFile('PeterNgTr');
```

Reply:
amazing proposal. I made the changes.
- now it returns the `fileName`
- moreover if the custom name is provided, it's used for downloaded file, otherwise default one is used.